src/BackendStatusBanner/index.jsx

```diff
@@ -1,39 +1,41 @@
 import React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+
 import statusMap from './statusMap.json';
+import { pingStudio } from '../data/actions/pingStudio';
 
 class BackendStatusBanner extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      apiConnectionStatus: 200,
-    };
-  }
-
   componentDidMount() {
-    fetch('/api/assets/course-v1:edX+DemoX+Demo_Course/?page=0&page_size=50&sort=sort&asset_type=', {
-      credentials: 'same-origin',
-      headers: {
-        Accept: 'application/json',
-      },
-    })
-      .then((response) => {
-        if (response.status !== this.state.apiConnectionStatus) {
-          this.setState({
-            apiConnectionStatus: response.status,
-          });
-        }
-      });
+    this.props.pingStudio();
   }
 
   render() {
-    return (this.apiConnectionStatus === 200) ?
+    return (this.props.connectionStatus === 200) ?
       null :
       (
         <div className="api-error">
-          {statusMap[this.state.apiConnectionStatus]}
+          {statusMap[this.props.connectionStatus]}
         </div>
       );
   }
 }
 
-export default BackendStatusBanner;
+BackendStatusBanner.propTypes = {
+  connectionStatus: PropTypes.number,
+  pingStudio: PropTypes.func.isRequired,
+};
+
+BackendStatusBanner.defaultProps = {
+  connectionStatus: null,
+};
+
+const WrappedBackendStatusBanner = connect(
+  state => ({
+    connectionStatus: state.connectionStatus,
+  }), dispatch => ({
+    pingStudio: () => dispatch(pingStudio()),
```

Comment:
This is more a question for my own understanding than a comment on the code. Is this mapping the pingStudio function that's being used at this [line](https://github.com/edx/studio-frontend/pull/15/files#diff-7fadfae4565e9c1514e2a49e17a2288cR17) to the function dispatch(pingStudio())? Is the function considered a thunk - not sure if I'm using thunk correctly in this context, since I'm leaning a bit on my knowledge of Scheme? Is its execution delayed until it's used from the aforementioned line 17?

Reply:
@MichaelRoytman exactly. The arrow function `() => dispatch(pingStudio())` wraps the invocation of the action creator `pingStudio` (the one we import at the top of the file) in a function. So when this function is called, it invokes `pingStudio` and passes its result to `dispatch`. Like you said, it only gets called where the prop function is invoked on line 17. 

One thing I want to point out is that the prop name `pingStudio` (the key in the object containing this function) does *not* have to match the name of the action creator. We could do
```
foo: () => dispatch(pingStudio()),
```
and then within our component we'd invoke `this.props.foo()`, and it would do the same thing. I just called it `pingStudio` to be consistent.